src/main/java/com/microsoft/azure/vmagent/builders/AdvancedImageFluent.java

```diff
@@ -65,11 +67,13 @@ public T withCustomImage(String imageUrl) {
         return (T) this;
     }
 
-    public T withReferenceImage(String imagePublisher,
+    public T withReferenceImage(String imageId,
+                                String imagePublisher,
                                 String imageOffer,
                                 String imageSku,
                                 String imageVersion) {
         this.imageReferenceType = "reference";
+        this.imageId = imageId;
```

Comment:
`imageId` should not in the `withReferenceImage()`. Please add a new function maybe `withCustomManagedImage()` and  set `this.imageReferenceType = "customImage"`
And thus you need to do some change in `AdvancedImageBuilder`.